src/main/java/org/broadinstitute/hellbender/engine/FeatureInput.java

```diff
@@ -60,6 +58,11 @@
     private final String featureFile;
 
     /**
+     * Cache the codec for this feature input the first time we discover it, so we only do it once
+     */
+    private transient Class<FeatureCodec<T, ?>> featureCodecClass;
```

Comment:
Let me take a look to see if it would be possible to do this at the `FeatureDataSource` level rather than the `FeatureInput` level.